client/src/main/java/org/apache/nemo/client/JobLauncher.java

```diff
@@ -259,7 +262,9 @@ public static void launchDAG(final IRDAG dag,
 
     LOG.info("Launching DAG...");
     serializedDAG = Base64.getEncoder().encodeToString(SerializationUtils.serialize(dag));
-    jobDoneLatch = new CountDownLatch(1);
+    if (jobDoneLatch == null || jobDoneLatch.getCount() == 0) {
```

Comment:
Why do we need to check latch value here?
Wouldn't setup be called earlier than launchDAG?

Reply:
Before, it just simply re-initialized the jobDoneLatch no matter what. I wanted to double-check that it was initialized only when it was necessary.